Algorithms/Recursion_Algorithm/Permutation.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LeetCode46
+{
+    class Permutations
+    {
+        public static List<string> SinglePermutations(string s)
+        {
+			// Example: s = "abc"
+			
+			// Set is used to avoid duplicates
+            var hash = new SortedSet<string>();
+            var stack = "";
+            int len = s.Length;
+            GetPermutations(s, hash, stack, len);
+            return new List<string>(hash);
+			
+			// returned value is ["abc", "acb", "bac", "bca", "cab", "cba"] 
+        }
+
+        public static void GetPermutations(string item, SortedSet<string> hash, string stack, int len)
```

Comment:
@Danotsonof IMHO, `GetPermutations` should be a private method.

Reply:
@ajeur

> @Danotsonof IMHO, `GetPermutations` should be a private method.

Thanks, done.